models/src/main/kotlin/io/provenance/api/models/cee/execute/ContractExecutionResponse.kt

```diff
@@ -1,28 +1,43 @@
 package io.provenance.api.models.cee.execute
 
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.fasterxml.jackson.annotation.JsonTypeName
 import io.provenance.api.models.p8e.TxResponse
 import java.util.UUID
 
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonSubTypes(
+    JsonSubTypes.Type(value = SinglePartyContractExecutionResponse::class, name = "single"),
+    JsonSubTypes.Type(value = MultipartyContractExecutionResponse::class, name = "multi"),
+    JsonSubTypes.Type(value = ContractExecutionErrorResponse::class, name = "error"),
+)
 sealed class ContractExecutionResponse(
     open val scopeUuids: List<UUID>,
     open val error: String? = null,
     val multiparty: Boolean? = null,
 )
 
+@JsonTypeName("single")
 data class SinglePartyContractExecutionResponse(
+    val type: String = "single",
     val metadata: TxResponse?,
     override val error: String? = null,
     override val scopeUuids: List<UUID>
 ) : ContractExecutionResponse(error = error, multiparty = false, scopeUuids = scopeUuids)
 
+@JsonTypeName("multi")
 data class MultipartyContractExecutionResponse(
+    val type: String = "multi",
```

Comment:
If the base class is already providing `multiparty` as a property, do we want to add `@JsonIgnore` to these? Do they need to be present in the output for this solution to work?

Reply:
Maybe the `multiparty` property is the obsolete part